src/main/java/org/opentripplanner/service/worldenvelope/model/WorldEnvelopeBuilder.java

```diff
@@ -0,0 +1,98 @@
+package org.opentripplanner.service.worldenvelope.model;
+
+import java.util.Collection;
+import java.util.function.Function;
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+
+/**
+ * Calculates lower/upper right/left latitude and longitude of all the coordinates.
+ * <p>
+ * This takes into account that envelope can extend over 180th meridian
+ */
+public class WorldEnvelopeBuilder {
+
+  private static final Double MIN_NOT_SET = 9999d;
+  private static final Double MAX_NOT_SET = -9999d;
+
+  private double minLat = MIN_NOT_SET;
+  private double maxLat = MAX_NOT_SET;
+  private double minLonWest = MIN_NOT_SET;
+  private double maxLonWest = MAX_NOT_SET;
+  private double minLonEast = MIN_NOT_SET;
+  private double maxLonEast = MAX_NOT_SET;
+
+  private WgsCoordinate transitMedianCenter = null;
+
+  public WorldEnvelopeBuilder expandToIncludeStreetEntities(double latitude, double longitude) {
+    return this.expandToInclude(latitude, longitude);
+  }
+
+  /**
+   * Calculates the center from median of coordinates of the elements in the given collection.
+   * <p>
+   * This speeds up calculation, but problem is that median needs to have all latitudes/longitudes
+   * in memory, this can become problematic in large installations. It works without a issues on New
+   * York State.
+   */
+  public <T> WorldEnvelopeBuilder expandToIncludeTransitEntities(
+    Collection<T> collection,
+    Function<T, Double> latProvider,
+    Function<T, Double> lonProvider
+  ) {
+    if (collection.isEmpty()) {
+      return this;
+    }
+
+    // Expand Envelope
+    for (T it : collection) {
+      expandToInclude(latProvider.apply(it), lonProvider.apply(it));
+    }
+
+    // we need this check because there could be only AreaStops (which don't have vertices)
+    // in the graph
```

Comment:
What does this refer to, I don't see anything related to vertices?

Reply:
Nor do I, I went back and the same comment is in the old code witch is unchanged, it is just inlined. I think I will delete it. Should we filter the vertexes when calculating the bounding box? I do not do it now, and in the old code it did all Stop vertexes first, then ist added Entrence, Pathways ... ;-). The performance is not a problem, so it is strictly for functional purposes. I assume all vertexes have a valid coordinate.

Reply:
Seems we lost the check [here](https://github.com/opentripplanner/OpenTripPlanner/commit/3f9e1f34#diff-b88fdbdbe8162c01cf8c60728f20d823f7f63986ba739ed5fbe8d37bd8a0b531L249)